machine-learning/lambda-v2/create_components.py

```diff
@@ -0,0 +1,387 @@
+            except Exception as e:
+                print(
+                    "Failed to create the component using the recipe at {}.\nException: {}".format(
+                        recipe_file_path, e
+                    )
+                )
+                exit(1)
+            print("Created component {}".format(component_recipe))
+
+
+def create_lambda_components():
+    for component_recipe in os.listdir(build_lambda_recipes_path):
+        recipe_file_path = os.path.join(build_lambda_recipes_path, component_recipe)
+        with open(
+            recipe_file_path,
+        ) as f:
+            try:
+                greengrass_client.create_component_version(lambdaFunction=json.load(f))
+            except Exception as e:
+                print(
+                    "Failed to create the component using the recipe at {}.\nException: {}".format(
+                        recipe_file_path, e
+                    )
+                )
+                exit(1)
+            print("Created lambda component {}".format(component_recipe))
+
+
+def get_account_number():
+    try:
+        response = sts_client.get_caller_identity()
+        if response is not None:
+            account = response["Account"]
+            return account
+    except Exception as e:
+        print("Cannot get the account id from the credentials.\nException: {}".format(e))
+        exit(1)
+
+
+def skip_creation(component):
+    return (
+        (component_argument != "" and component_argument != component)
+        or component.startswith(".")
+        or inferenceType not in component
+    )
+
+
+def skip_lambda_component_creation(component):
+    return skip_creation(component) or (
+        containerMode.lower() == "true" and ".Container" not in component
+    )
+
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+artifacts_path = os.path.join(dir_path, "artifacts")
+lambda_artifacts_path = os.path.join(dir_path, "lambda-artifacts")
+recipes_path = os.path.join(dir_path, "recipes")
+lambda_recipes_path = os.path.join(dir_path, "lambda-recipes")
+build_dir_path = os.path.join(dir_path, "build")
+build_artifacts_path = os.path.join(build_dir_path, "artifacts")
+build_lambda_artifacts_path = os.path.join(build_dir_path, "lambda-artifacts")
+build_recipes_path = os.path.join(build_dir_path, "recipes")
+build_lambda_recipes_path = os.path.join(build_dir_path, "lambda-recipes")
+
+shutil.rmtree(build_dir_path, ignore_errors=True, onerror=None)
+
+region = "us-east-1"
+bucket = "ggv2-example-component-artifacts"
+inferenceType = ""
+component_argument = ""
+model_releases = (
+    "https://github.com/aws-greengrass/aws-greengrass-component-examples/releases/download/v1.0"
+)
+
+# Parse the arguments
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "--region",
+    "-r",
+    default=region,
+    help="Greengrass components will be created in the region.",
+)
+parser.add_argument(
+    "--bucket",
+    "-b",
+    default=bucket,
```

Comment:
why should we have a default bucket? Bucket names are globally unique, our default won't work for anybody.

Reply:
this bucket is for storing the artifacts. We create one for region if it doesn't exist already(Mentioned in README that we append region name to the bucket name). Customers are free to choose their bucket. Do you still see a concern?

Reply:
Yes. There should not be a default because the default won't work for the second customer who tries it. They are globally unique, not by account and not by region, but global.